frontend/ui/uimanager.lua

```diff
@@ -749,6 +748,18 @@ function UIManager:quit(exit_code)
         self.looper:close()
         self.looper = nil
     end
+    return self._exit_code
+end
+
+-- A simple wrapper for UIManager:quit()
+-- This may be overwritten by setRunForeverMode(); for testing purposes
+function UIManager:_gated_quit(exit_code)
+    return self:quit(exit_code)
+end
+
+-- Disable automatic UIManager quit; for testing purposes
+function UIManager:setRunForeverMode()
+    self._gated_quit = function() return false end
```

Comment:
I don't recall if the testsuite ever unloads UIManager, so, we *may* need a way to restore the standard wrapper, too?

Reply:
The test suite does a `package.unloadAll()` on most (all?) setup functions. On my laptop and CircleCi finish without error. 
So the answer is probably: No, we don't _need_ to have a restore functionallity.
(But I will implement these three lines for completeness.)